USBPcapDriver/USBPcapURB.c

```diff
@@ -673,41 +673,52 @@ VOID USBPcapAnalyzeURB(PIRP pIrp, PURB pUrb, BOOLEAN post,
             }
             packetHeader->header.transfer = USBPCAP_TRANSFER_ISOCHRONOUS;
 
+            transferLength = 0;
             if (transfer->TransferFlags & USBD_TRANSFER_DIRECTION_IN)
             {
                 if (post == TRUE)
                 {
                     /* Read from device, return from controller */
-                    attachData = TRUE;
-                }
-                else
-                {
-                    /* Read from device, on its way to controller */
-                    attachData = FALSE;
+                    if (transfer->TransferBuffer)
+                    {
+                        /* return packet buffers only if data has been transferred */
+                        if (transfer->TransferBufferLength > 0)
+                        {
+                            for (i = 0; i < transfer->NumberOfPackets; i++)
+                            {
+                                /* capture only packet buffers that contain data to minimize capture size */
```

Comment:
To really minimize capture size, it would be necessary to recombine the data to leave out the gaps (which would require quite a bit more code). Maybe change the comment to something like "determine how many transfer buffer bytes to capture in order to include data from all nonempty packets"?